src/python/tests/core/test_scheduler.py

```diff
@@ -0,0 +1,212 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+import asyncio
+import datetime
+
+import pytest
+import uuid
+
+from foglamp.core.scheduler import Scheduler, IntervalSchedule, ScheduleNotFoundError, Task
+
+
+__author__ = "Terris Linenbach"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+class TestScheduler:
+    @staticmethod
+    async def stop_scheduler(scheduler: Scheduler)->None:
+        while True:
+            try:
+                await scheduler.stop()
+                break
+            except TimeoutError:
+                await asyncio.sleep(1)
+
+    @pytest.mark.asyncio
+    async def test_stop(self):
+        scheduler = Scheduler()
+
+        await scheduler.populate_test_data()
+        await scheduler.start()
+
+        interval_schedule = IntervalSchedule()
+        interval_schedule.exclusive = False
+        interval_schedule.name = 'sleep1'
+        interval_schedule.process_name = "sleep1"
+        interval_schedule.repeat = datetime.timedelta(seconds=1)
+
+        await scheduler.save_schedule(interval_schedule)
+        await asyncio.sleep(10)
+
+        await self.stop_scheduler(scheduler)
+
+    @pytest.mark.asyncio
+    async def test_save(self):
+        scheduler = Scheduler()
+
+        await scheduler.populate_test_data()
+        await scheduler.start()
+
+        interval_schedule = IntervalSchedule()
+        interval_schedule.name = 'sleep1'
+        interval_schedule.process_name = "sleep1"
+        interval_schedule.repeat = datetime.timedelta(seconds=1)
+
+        await scheduler.save_schedule(interval_schedule)
+
+        # TODO: Re-read the task and check the type (need API support)
+
+        await asyncio.sleep(10)
+
+        # TODO: check for task created (need API support)
```

Comment:
now we have API support; still want to keep this as TODO?! for assert. Just an ask!